rpc/handlers.go

```diff
@@ -1169,6 +1181,31 @@ func (h *Handler) AddTransaction(ctx context.Context, tx BroadcastedTransaction)
 	}, nil
 }
 
+func (h *Handler) addToMempool(tx *BroadcastedTransaction) (*AddTxResponse, *jsonrpc.Error) {
+	userTxn, userClass, _, err := adaptBroadcastedTransaction(tx, h.bcReader.Network())
+	if err != nil {
+		return nil, ErrInternal.CloneWithData(err.Error())
+	}
+	err = h.memPool.Push(&mempool.BroadcastedTransaction{
+		Transaction:   userTxn,
+		DeclaredClass: userClass,
+	})
+	if err != nil {
+		return nil, ErrValidationFailure.CloneWithData(err.Error())
```

Comment:
The error could be an internal database error. Maybe we should have `mempool.ErrInvalid` to know when to return `ErrInternal` and `ErrValidationFailure`.

Reply:
similar to https://github.com/NethermindEth/juno/pull/1622#discussion_r1439200956